### PR TITLE
Add search filter

### PR DESCRIPTION
**Pull request by cazterk**

### CODE REVIEW

File: itez/beneficiary/views.py [hunk 1]
```diff
@@ -143,22 +152,54 @@ class BenenficiaryListView(LoginRequiredMixin, ListView):
     paginate_by = 10
     template_name = "beneficiary/beneficiary_list.html"
 
+    # print("------------------------------>", context["qs_json"])
```
Comment (sonlinux) [inline itez/beneficiary/views.py]: lets remove this

File: itez/beneficiary/views.py [hunk 2]
```diff
@@ -143,22 +152,54 @@ class BenenficiaryListView(LoginRequiredMixin, ListView):
...
+
     def get_queryset(self):
-        return Beneficiary.objects.filter(alive=True)
+
+        if 'q' in self.request.GET:
+            q = self.request.GET['q']
+            beneficiary = Beneficiary.objects.filter(Q(first_name__contains=q) | Q(
```
Comment (sonlinux) [inline itez/beneficiary/views.py]: `Beneficiary.objects.filter(alive=True and Q(first_name__contains=q)..`

File: itez/beneficiary/views.py [hunk 3]
```diff
@@ -143,22 +152,54 @@ class BenenficiaryListView(LoginRequiredMixin, ListView):
...
     def get_context_data(self, **kwargs):
         context = super(BenenficiaryListView, self).get_context_data(**kwargs)
+        beneficiary_resource = BeneficiaryResource()
+        export_data = beneficiary_resource.export()
+        export_type = export_data.json
+
+        # fn = u.all
+        # s = r.post[]
+        # for i in fn
+        #     if i.startswith(s)
```
Comment (sonlinux) [inline itez/beneficiary/views.py]: lets clean this  up

File: itez/beneficiary/views.py [hunk 4]
```diff
@@ -143,22 +152,54 @@ class BenenficiaryListView(LoginRequiredMixin, ListView):
...
         context["vl"] = Service.objects.filter(service_type="VL").count()
         context["art"] = Service.objects.filter(client_type="ART").count()
         context["labs"] = Service.objects.filter(service_type="LAB").count()
         context["pharmacy"] = Service.objects.filter(service_type="PHARMACY").count()
         context["title"] = "Beneficiaries"
+
+        # beneficiary = Beneficiary.objects.get.all(beneficiary_id=pk_test)
+
+        # beneficiaries = beneficiary_filter.qs
+
```
Comment (sonlinux) [inline itez/beneficiary/views.py]: lets remove all commented code

File: itez/beneficiary/views.py [hunk 5]
```diff
@@ -143,22 +152,54 @@ class BenenficiaryListView(LoginRequiredMixin, ListView):
...
+        context["qs_json"] = export_type  # [value for value in Beneficiary.objects.values()]
```
Comment (sonlinux) [inline itez/beneficiary/views.py]: lets call this `search_json_qs`

File: itez/beneficiary/views.py [hunk 6]
```diff
@@ -143,22 +152,54 @@ class BenenficiaryListView(LoginRequiredMixin, ListView):
...
+
         return context
 
+    def search(sleft, request, args, **kwargs):
+        beneficiaries = Beneficiary.objects.all()
+        beneficiary_filter = BeneficiaryFilter(request.GET, queryset=beneficiaries)
+        return render(request, {'filter': beneficiary_filter})
```
Comment (sonlinux) [inline itez/beneficiary/views.py]: this can go

File: itez/beneficiary/views.py [hunk 7]
```diff
@@ -180,7 +221,7 @@ def get_context_data(self, **kwargs):
         return context
 
 
-@login_required(login_url="/login/")
+@ login_required(login_url="/login/")
```
Comment (sonlinux) [inline itez/beneficiary/views.py]: this need to be `@login_required(login_url="/login/")` and not `@ login_required(login_url="/login/")`

File: itez/beneficiary/views.py [hunk 8]
```diff
@@ -194,7 +235,7 @@ def user_events(request):
     return HttpResponse(html_template.render(context, request))
 
 
-@login_required(login_url="/login/")
+@ login_required(login_url="/login/")
```
Comment (sonlinux) [inline itez/beneficiary/views.py]: same here

File: itez/beneficiary/views.py [hunk 9]
```diff
@@ -217,7 +258,7 @@ def beneficiary_report(request):
...
```
Comment (sonlinux) [inline itez/beneficiary/views.py]: do the same here

File: itez/templates/beneficiary/beneficiary_list.html [hunk 10]
```diff
@@ -67,9 +67,19 @@ <h4 class="card-title card-title-dash">All Beneficiaries</h4>
...
+                                <input type="search" class="form-control" name="q" placeholder="Search beneficiary here" title="Search here" id="search_here">
                                 <!-- <i class="icon-search"></i> -->
+                                <button class="btn btn-primary" type="submit">Search</button>
                               </form>
+                              <div id="box">
+                                {% for item in object_list %}   
+                                  <!-- <b>{{item}}</b> <br> -->
+                                {% endfor %}
+
+                                </div>
```
Comment (sonlinux) [inline itez/templates/beneficiary/beneficiary_list.html]: we dont need to print suggestions on the table section 
instead they should be highlights

File: itez/templates/beneficiary/beneficiary_list.html [hunk 11]
```diff
@@ -220,4 +230,37 @@ <h3>No Beneficiaries Available</h3>
   <script src="/static/assets/js/dashboard.js"></script>
   <script src="/static/assets/js/Chart.roundedBarCharts.js"></script>
 
+  <script>
+    const data = '{{qs_json}}'
```
Comment (sonlinux) [inline itez/templates/beneficiary/beneficiary_list.html]: lets update this with context variable too

File: itez/templates/beneficiary/beneficiary_list.html [hunk 12]
```diff
@@ -220,4 +230,37 @@ <h3>No Beneficiaries Available</h3>
...
+    let filterredArr = []
+
+    input.addEventListener('keyup', (e)=>{
+      box.innerHTML = ""
+      filterredArr = rdata.filter(info=>info['first_name'].includes(e.target.value) )
+      console.log("filted-array :",filterredArr)
+
+      if(filterredArr.length > 0){
+        filterredArr.map(info=>{
+          box.innerHTML += `<b> ${info['first_name']}  ${info['last_name']}</b><br>`
```
Comment (sonlinux) [inline itez/templates/beneficiary/beneficiary_list.html]: we need to include `beneficiary ID` too

File: requirements/base.txt [hunk 13]
```diff
@@ -23,6 +23,7 @@ django-role-permissions==3.1.1 # https://github.com/vintasoftware/django-role-pe
 django-imagekit==4.1.0 # https://github.com/matthewwithanm/django-imagekit
 django-import-export==2.5.0
 djangocms-admin-style==2.0.2
+django-filter==21.1 
 # Common
```
Comment (sonlinux) [inline requirements/base.txt]: leave a space below here

**Review by sonlinux — CHANGES_REQUESTED** [review]

File: itez/beneficiary/views.py (hunk 1 above)
Comment (cazterk) [reply itez/beneficiary/views.py]: - [x] remove commented code

File: itez/beneficiary/views.py (hunk 2 above)
Comment (cazterk) [reply itez/beneficiary/views.py]: - [x] add alive condition

File: itez/beneficiary/views.py (hunk 3 above)
Comment (cazterk) [reply itez/beneficiary/views.py]: - [x] remove commented code

File: itez/beneficiary/views.py (hunk 4 above)
Comment (cazterk) [reply itez/beneficiary/views.py]: - [x] remove commented code

File: itez/beneficiary/views.py (hunk 6 above)
Comment (cazterk) [reply itez/beneficiary/views.py]: - [x] remove redundant code

File: itez/beneficiary/views.py (hunk 5 above)
Comment (cazterk) [reply itez/beneficiary/views.py]: - [x] 'qs-json' => 'search_json_qs'

File: itez/beneficiary/views.py (hunk 7 above)
Comment (cazterk) [reply itez/beneficiary/views.py]: - [x] delete initial space

File: itez/beneficiary/views.py (hunk 8 above)
Comment (cazterk) [reply itez/beneficiary/views.py]: - [x] delete initial space

File: itez/beneficiary/views.py (hunk 9 above)
Comment (cazterk) [reply itez/beneficiary/views.py]: - [x] delete initial space

File: requirements/base.txt (hunk 13 above)
Comment (cazterk) [reply requirements/base.txt]: - [x] leave space

File: itez/templates/beneficiary/beneficiary_list.html (hunk 12 above)
Comment (cazterk) [reply itez/templates/beneficiary/beneficiary_list.html]: - [x] add beneficiary_id

File: itez/templates/beneficiary/beneficiary_list.html (hunk 11 above)
Comment (cazterk) [reply itez/templates/beneficiary/beneficiary_list.html]: - [x] update context

File: itez/templates/beneficiary/beneficiary_list.html (hunk 10 above)
Comment (cazterk) [reply itez/templates/beneficiary/beneficiary_list.html]: - [x] remove print suggections

**Review by sonlinux — APPROVED** [review]

**Pull request merged**